src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java

```diff
@@ -0,0 +1,78 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.integ_test;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.integ_test.json.JsonResponseData;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.integ_test.json.JsonResponseField;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.integ_test.json.JsonResponseNode;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.function.DoublePredicate;
+
+/**
+ * All the heap tests check the metrics are within a range.
+ * We can't check for absolute values as it'll vary across runs.
+ */
+public class HeapMetricsIT  extends MetricCollectorIntegTestBase {
+
+    private static final Logger LOG = LogManager.getLogger(HeapMetricsIT.class);
+    private double MIN_HEAP = 50 * 1024 * 1024 ; // 50 MB
+    private double MAX_HEAP = 512 * 1024 * 1024 ; // 512 MB
+    @Before
+    public void init() throws Exception {
+        initNodes();
+    }
+
+    @Test
+    public void checkHeapInit() throws Exception {
+        checkHeapMetric(AllMetrics.HeapValue.HEAP_INIT, (d) -> d >= MIN_HEAP && d <= MAX_HEAP);
+    }
+
+    @Test
+    public void checkHeapMax() throws Exception {
+        checkHeapMetric(AllMetrics.HeapValue.HEAP_MAX, (d) -> d >= MIN_HEAP && d <= MAX_HEAP);
```

Comment:
is HeapMax supposed to be equivalent to MAX_HEAP ?

Reply:
It'll depend on max memory allocated to elastic search JVM. In the current set up we have provided 512 MB for JVM.  I kept the check as a range, if we run on a environment with lower memory.